rules/app.bzl

```diff
@@ -56,7 +57,16 @@ def ios_application(name, apple_library = apple_library, **kwargs):
         apple_library: The macro used to package sources into a library.
         **kwargs: Arguments passed to the apple_library and ios_application rules as appropriate.
     """
-    infoplists = write_info_plists_if_needed(name = name, plists = kwargs.pop("infoplists", []))
+
+    infoplists_by_build_setting = kwargs.pop("infoplists_by_build_setting", {})
+    for (build_setting, plists) in infoplists_by_build_setting.items():
+        name_suffix = build_setting_name(build_setting)
+        infoplists_by_build_setting[build_setting] = write_info_plists_if_needed(name = "%s.%s" % (name, name_suffix), plists = plists)
+
+    infoplists_by_build_setting["//conditions:default"] = write_info_plists_if_needed(name = name, plists = kwargs.pop("infoplists", []))
```

Comment:
will this stomp a value that's explicitly passed in for `"//conditions:default"`? If so, should we handle that case?

Reply:
That's true, since we're accepting any incoming `infoplists_by_build_setting` it makes sense to respect a existing default value if it exists, I'll add the check